### ISSUE TITLE
NO SCHEMA AVAILABLE when setting a UNIQUE constraint on a specific TEXT column

### ISSUE THREAD

**Issue by eliecerthoms** [graphile/postgraphile#767]
I'm submitting a ... 

* [x] bug report
* [ ] feature request
* [ ] question 

PostGraphile version:

4.0.0-beta.9

Minimal SQL file that can be loaded into a clean database:

```sql
CREATE TABLE crtrs."user" (
	first_name text NOT NULL, -- The user's first names
	last_name text NOT NULL, -- The user's last names
	created_at timestamptz NULL DEFAULT timezone('america/bogota'::text, now()), -- The time the user was created
	updated_at timestamptz NULL DEFAULT timezone('america/bogota'::text, now()), -- The time the user information was last updated
	id uuid NOT NULL DEFAULT gen_random_uuid(), -- The user identifier
	slug text NOT NULL,
	profile_pic text NULL,
	CONSTRAINT user_slug_key UNIQUE (slug),
	CONSTRAINT usuario_apellido_check CHECK ((char_length(last_name) < 100)),
	CONSTRAINT usuario_nombre_check CHECK ((char_length(first_name) < 100)),
	CONSTRAINT usuario_pk PRIMARY KEY (id)
)
WITH (
	OIDS=FALSE
) ;
CREATE UNIQUE INDEX usuario_id_idx ON crtrs."user" USING btree (id) ;
COMMENT ON TABLE crtrs."user" IS 'A user of the app' ;
```

Steps to reproduce:

I'm running the library version... I don't think I have written any code that has any relation to this issue

Current behavior:
In the instant I add a UNIQUE constraint to the `slug` column, postgraphile breaks completely. The graphiql interface shows NO SCHEMA AVAILABLE and the terminal shows hundreds of errors in this fashion:
```
The type of DeleteUserPayload.query must be Output Type but got: undefined.
```
Basically one for every mutation.

Expected behavior:

It is expected for this issue not to happen. The weird thing is that I've tried adding a UNIQUE constraint to a lot of other columns and the error doesn't happen. It happens ONLY when I add it to the `slug` column :s

**Comment by benjie** [comment]
That is super strange. It looks like a “multiple versions of GraphQL” issue but I don’t understand why that particular issue would trigger it.

Can you “find -name graphql -type d” inside your project and send the results please?

**Comment by benjie** [comment]
Have you changed the order of the internal plugins at all? What your error implies is that this plugin:

https://github.com/graphile/graphile-build/blob/master/packages/graphile-build/src/plugins/MutationPayloadQueryPlugin.js

Fails to get a reference to the Query type.

**Comment by eliecerthoms** [comment]
Nope, I haven't tinkered with plugins whatsoever. What type of information should the command you told me return? I'm using windows and therefore the `dir` command since `find` apparently only works for files here. 

This is what `dir` returns (translated from spanish)

```
C:\Users\user\Desktop\project\server\socket (master -> origin) (server@1.0.0)
λ dir graphql /s
 C:\Users\user\Desktop\project\server\socket\node_modules

05/13/2018  01:50 p. m.    <DIR>          graphql
              0 files 0 bytes

C:\Users\user\Desktop\project\server\socket\node_modules\@types

05/13/2018  01:04 p. m.    <DIR>          graphql
               0 files 0 bytes

     Total files in the list:
               0 files 0 bytes
               2 dirs  76,122,738,688 bytes free
```

Apparently there is a `graphql` folder in both `node_modules` and `@types`

**Comment by benjie** [comment]
That’s fine. Hmmm, no idea why it’d be then. Does the same thing happen when you install postgraphile into a clean directory and run it using the CLI?

**Comment by eliecerthoms** [comment]
No luck still. Gonna try to upgrade to pg10

**Comment by benjie** [comment]
SQL file that can be loaded into an empty DB:

```sql
create schema crtrs;
create extension "uuid-ossp";
create extension "pgcrypto";
CREATE TABLE crtrs."user" (
	first_name text NOT NULL, -- The user's first names
	last_name text NOT NULL, -- The user's last names
	created_at timestamptz NULL DEFAULT timezone('america/bogota'::text, now()), -- The time the user was created
	updated_at timestamptz NULL DEFAULT timezone('america/bogota'::text, now()), -- The time the user information was last updated
	id uuid NOT NULL DEFAULT gen_random_uuid(), -- The user identifier
	slug text NOT NULL,
	profile_pic text NULL,
	CONSTRAINT user_slug_key UNIQUE (slug),
	CONSTRAINT usuario_apellido_check CHECK ((char_length(last_name) < 100)),
	CONSTRAINT usuario_nombre_check CHECK ((char_length(first_name) < 100)),
	CONSTRAINT usuario_pk PRIMARY KEY (id)
)
WITH (
	OIDS=FALSE
) ;
CREATE UNIQUE INDEX usuario_id_idx ON crtrs."user" USING btree (id) ;
COMMENT ON TABLE crtrs."user" IS 'A user of the app' ;
```

**Comment by benjie** [comment]
Created database `i767` using the commands above.

Then I ran this:

```
postgraphile -s crtrs -c postgres://localhost/i767 --watch
```

Then this SQL:

```sql
alter table crtrs."user" add constraint uniq_user_slug unique(slug);
```

GraphiQL updates as expected:

<img width="351" alt="screenshot 2018-05-27 15 04 55" src="https://user-images.githubusercontent.com/129910/40586818-5473a262-61bf-11e8-9d3a-e119db80af93.png">

I'll need better reproduction steps to help further.

**Comment by eliecerthoms** [comment]
Strange indeed.  I even upgraded to Postgres10 and that constraint is **apparently** still breaking the schema. Oh well, going to have to do without it. I think this can be closed now.
